File: chap1/Poste.js
// Ces programmes sont sous licence CeCILL-B V1.
var type; 
var poids;
type = ISNinputString ();
poids = ISNinputInt (); 
if (type=="verte") {
	if (poids <= 20) {
		ISNprintln('0,57 euro');
	}
		else if (poids <= 50) {
			ISNprintln('0,95 euro');
		}
			else if (poids <= 100) {
				ISNprintln('1,40 euro');
			}			
}
	else if (type=="prioritaire") {
		if (poids <= 20) {
			ISNprintln('0,60 euro');
		}
			else if (poids <= 50) {
				ISNprintln('1,00 euro');
			}
				else if (poids <= 100) {
					ISNprintln('1,45 euro');
				}
	}
		else if (type=="ecopli") {
			if (poids <= 20) {
				ISNprintln('0,55 euro');
			}
				else if (poids <= 50) {
					ISNprintln('0,78 euro');
				}
					else if (poids <= 100) {
						ISNprintln('1,00 euro');
					}
		}

File: chap1/Poste.cpp
// Ces programmes sont sous licence CeCILL-B V1.
#include <iostream>
#include <string>

using namespace std;

int main()
{
   string type; 
   int poids;
   cin >> type;
   cin >> poids; 
   if (type.compare("verte") == 0)
   {
      if (poids <= 20)
      {
         cout << 0.57 << endl;
      }
      else 
      {
         if (poids <= 50)
         {
            cout << 0.95 << endl;
         }
         else
         {
            if (poids <= 100)
            {
               cout << 1.40 << endl;
            }
         }
      }
   }
   else
   {
      if (type.compare("prioritaire") == 0)
      {
         if (poids <= 20)
         {
            cout << 0.60 << endl;
         }
         else
         {
            if (poids <= 50)
            {
               cout << 1.00 << endl;
            }
            else
            {
               if (poids <= 100)
               {
                  cout << 1.45 << endl;
               }
            }
         }
      }
      else 
      {
         if (type.compare("ecopli") == 0) 
         {
            if (poids <= 20) 
            {
               cout << 0.55 << endl;
            }
            else 
            {
               if (poids <= 50)
               {
                  cout << 0.78 << endl;
               }
               else 
               {
                  if (poids <= 100) 
                  {
                     cout << 1.00 << endl;
                  }
               }
            }
         }
      }
   }
}
